### ISSUE TITLE
Add check of result chmod() in RAND_file_name(...)

### ISSUE THREAD

**Issue by nv-dmd** [openssl/openssl#25143]
OpenSSL 3.0.14

In the function RAND_file_name(), chmod(file, 0600) is executed on line 249.
But the result of the chmod(...) function execution is not checked.
Maybe it is worth checking the result, because chmod(...) should restrict permissions and its successful execution seems important?

**Comment by paulidale** [comment]
Yes, it is import **but only if you are running VMS**.  On Unix like systems, the file is created with mode _0600_ on line 210 making this redundant.

This `chmod` should be conditioned as VMS only and the return value should be checked.
It's also a race condition waiting to happen.

**Comment by JohnnySavages** [comment]
> Yes, it is import but only if you are running VMS.

What about openssl_fopen line 238 call? This case does not set any permission except default.

**Comment by levitte** [comment]
Something of note is that `open()` accepts a third argument on VMS, nowadays: https://docs.vmssoftware.com/vsi-c-run-time-library-reference-manual-for-openvms-systems/#OPEN_ROUTINE
So it seems to me that a lot of this code can be simplified.

**Comment by t8m** [comment]
Please note we are looking for community help implementing this issue. Please comment if you are interested in fixing the bug or if you have reasons why this bug is important enough so we should consider implementing it with higher priority than other requests. The issue will be closed after 2 months otherwise.

**Comment by t8m** [comment]
This issue was closed because it was marked with the help wanted label and there was no interest in implementing this issue or any other activity leading to implementing or re-prioritizing it for more than 2 months.
